### ISSUE TITLE
Performance: Short freezes/lag if GM kept open for a while after editing code

### ISSUE THREAD

**Issue by Retrific** [YoYoGames/GameMaker-Bugs#2528]
### Description

It's still the exact same issue as reported here:
https://github.com/YoYoGames/GameMaker-Bugs/issues/1650

When I have a (fairly big) project open for a couple of hours at some point I start noticing short freezes to the workspace (or entire IDE, not sure) every time I make edits, or rather a second or two after I stop making an edit. There is literally no difference between what I was experiencing in 2023.8 and now in 2023.11.

My PC is not maxed out, so its not an issue with my system resources. Only using 30% of my RAM and 10% of my CPU right now, but still experiencing these annoying stutters.

Check my comments there for more info.

Without being able to debug GM myself I cant tell you what causes this issue. But it only happens after a project is open for hours and restarting Game Maker fixes it.

### Expected Change

No more IDE freezes even when not restarting GM for days

### Steps To Reproduce

1. Start GameMaker
2. Open a project with at least like 30k lines of code (might not be important but its what I have). The amount of other assets are not important.
3. Dont close GM, maybe edit the project a bit, but mostly just let it sit there for at least 5 hours.
4. Type some code (and for better illustration move the workspace around)
5. Notice the freeze that happens shortly after being done editing

[contactme.zip](https://github.com/YoYoGames/GameMaker-Bugs/files/13705195/contactme.zip)

### How reliably can you recreate this issue using your steps above?

Always

### Which version of GameMaker are you reporting this issue for?

2023.11.1 (Monthly)

### Which operating system(s) are you seeing the problem on?

Windows 10

### Are you running GameMaker from inside your Steam library?

No

**Comment by nullbunny** [comment]
Can confirm... I am also working on a big project, and the IDE in general has gotten much more laggy when doing edits, the auto-complete box pops open, switching tabs... the problem seems to get worse the longer the IDE is open.

**Comment by RefresherTowel** [comment]
Just another user chiming in to say this is absolutely still a problem in GM. Constant freezing, stuttering, etc. If you're typing a line and the IDE lags, and you press enter, the section of text you typed while it was lagging gets "returned" to the next line, as though you actually pressed enter as soon as the lag started:

Like the lag starts here
but you pressed enter at the end of this line.

Many different issues like that causing chaos with text input, and problems when naming stuff after importing and all sorts of things. Makes it extremely frustrating to work with GM. I find myself having to pause and wait like every 30 seconds or so, because the lag is so frequent. It **heavily** reduces the amount of work you can get done in each session, and it's literally changed the way I interact with GM (I've been building anti-lag habits for so long now that it'll be actually hard to break them when the lag is finally fixed).

This is easily the most important bug fix GM requires, IMO, because everything else relies on the IDE being usable, which I would argue it isn't in its current state.

**Comment by nullbunny** [comment]
I can confirm the weird line break issue - I didn't add it here because I wasn't sure if it is me or part of the bug, but it is quite annoying.

**Comment by RhyminGarfunkle** [comment]
Also having this issue. [On the forums](https://forum.gamemaker.io/index.php?threads/2023-11-release-thread.107587/page-2#post-650584) it was recommended I post my system specs, so:
Windows 10
Motherboard: ASRock B450M
Processor: AMD Ryzen 5 5600X 
GPU: Radeon RX580
16gb RAM

I also use two monitors, the primary is 1440p with freesync, and the other is 1080p 60hz. I'm willing to post any other weird system specifics that might help.

**Comment by RefresherTowel** [comment]
My specs are

Windows 10
CPU: Intel(R) Core(TM) i7-7700 CPU @ 3.60GHz   3.60 GHz
RAM: 24gb
GPU: NVIDIA GeForce GTX 960
Motherboard: ASUSTek Prime H270M-Plus

Only a single monitor running at 1080p 60hz for me.

I always have both feather turned on and the inspector open, but it doesn't matter if I turn them off. I also get the lag regardless of project size (though it might be a little bit less with smaller projects, still frequent enough to make me unsure if it's actually less though).

**Comment by konjerk** [comment]
Windows 10 Pro x64
Intel(R) Core(TM) i9-9900K CPU @ 3.60GHz   3.60 GHz
32 GB RAM
GeForce RTX 3090
Two monitors (3440x1440 and 1920x1080) Game Maker only used on the larger one, have had other monitors during the issue

I have always had Feather off, I disable whatever that right-to-left read thing is, I toy with time to check syntax or suggest code and it still happens in some similar way. I have many tabs open per Object, and pretty large window sizes too. I don't use Inspector really. I get stagger at editing anything in 2023.8 and 2023.11, and slowdown over time which has been an issue of different size since after 2022.4 I think.

**Comment by KhaoMortadios** [comment]
I also want to comment on this, I'm genuinely _desperate_ for this to get fixed and want to help any way I can.

Windows 10
Motherboard : z170a gaming M3
CPU: Intel(R) Core(TM) i7-6700k @ 4.00GHz
GPU: NVIDIA GeForce GTX 1070
32GB RAM (DDR4, 2400 MHz)
I use 3 1920x1080 monitors.

If it helps, my PC has 2 SSDs, with both Game Maker and my projects being installed on the fastest one (which is also the same as the windows installation). I usually run Game Maker while running firefox, discord, Steam, github desktop, and 1-3 adobe programs (Animate, Photoshop or Premiere).

My biggest project contains 4000+ sprites and almost 500 objects, but I also get this issue on a smaller project with maybe 20 objects and something like 100 sprites.

**Comment by NanoSharkGames** [comment]
This issue is still happening to me as well on my machine.

macOS Sonoma 14.2.1
M2 Pro (12-core CPU, 19-core GPU)
16GB RAM

Feather is disabled. RAM usage starts at around half a GB & increases to 4GB after a couple hours of the IDE being opened & used. I spent most of the time editing code (without the inspector) & compiled the project around 10 times.

**Comment by PrismaticRealms** [comment]
I have a project with only a few hundred lines in it and just 2 small and simple sprite assets and 3 objects and a small room.  And I am experiencing the following issues:

- Several seconds to terminate a debug session.
- Several seconds to return to "Home" screen via the Home button.
- Several seconds to quit GameMaker IDE.
- Keystrokes randomly dropping (I type something and the character doesn't appear. Or I hit a function key, to step through the debugger for instance, and the key doesn't work all the time.)
- Small 1 or 2 second intermittent delays when moving the cursor around or typing text.

Using a pretty high-end Mac setup:
Mac Studio 2022 M1 Ultra, 20-core CPU, 48-core GPU
macOS Sonoma 14.2.1
64GB Memory

**Comment by nullbunny** [comment]
Also happens in the currently available Beta.

**Comment by Retrific** [comment]
Yea it might've even gotten worse in the new versions. In 2023.8 it took several hours before I started to notice the freezes. Now it doesnt even take 2 hours before it gets so annoying I either quit or have to restart the IDE.

This issue must get high priority, it really killed my enjoyment of working with Game Maker.
I dont need any new features, I need Yoyogames to focus on bugfixing.

**Comment by rwkay** [comment]
Please file a project (and ui.log) in fact a full bug as we get more information on your machine setup that way for this we are having difficulty replicating what you are seeing - it would be useful to know what you have been doing in a session that slows down...

We suspect it is a combination of the project (and what is in it) and what you are mainly working on (Room Editor, Code Editor, Image Editor)...

**Comment by Retrific** [comment]
Please check the attached file above. It has the ui.log and other files.
It also looks rather project independent to me when there are so many other people experiencing very similar issues. But I can send you a project privately.

Speaking for myself I am exclusively using code editor with only very little room editor changes inbetween. But it doesnt matter what I do because even if the project sits idle in the background and I am doing something entirely different or nothing at all and come back to Game Maker hours later it now started lagging.

**Comment by konjerk** [comment]
The test of la amount you can do where you edit code *at all* (like make just a space) then pull around the code window till you see it stagger, that is 100% tied to syntax check, at least for me. So it's somewhere in there. If I change check time to 5000ms the stagger happens after that time, etc. Game Maker also accrues RAM memory use over time. Is it somehow leaving what it's checked in memory at times? That would amass quickly. And again, I started getting issues after Feather was added.

This exact stagger thing hasn't been my only speed issue, I had issues starting after - I think - version 2022.4 after Feather was added, hence I have never had it on in case it helps. Back then it felt like things just generally slowed. Hard to tell if that was fully fixed with this other stagger thing. But yeah, it's become really, really bad in the latest public version, and I am once again gonna have to revert a few versions.

**Comment by KhaoMortadios** [comment]
Discovered something worth mentioning. I've had absolutely atrocious performance for months now, but today I decided to turn feather off.

All my performance issues were _completely_ gone. No stutters. Code editor works 100% smoothly. Some things are still unexpectedly slow (more notably, changing an object's sprite in my largest project) but navigating the IDE itself and editing code (easily what I do the most) has no performance issues from what I can see.

So personally, I'm completely willing to bet that the issue is 100% connected to feather.

**Comment by RhyminGarfunkle** [comment]
@KhaoMortadios I leave Feather disabled and still have stuttering when editing code and in the IDE.

It's definitely possible that they're linked, I first started seeing the stuttering issues about the time that it was released into the stable branch. But I don't see any changes in stuttering behavior with different Feather settings.

**Comment by KhaoMortadios** [comment]
@RhyminGarfunkle after using it a bit longer, you're right, the stutters came back. But it was still a clear improvement for me. So it's probably less of a feather-specific thing and more of a "the more things the IDE is trying to do at once the worse it gets". No idea if that's helpful.

**Comment by PrismaticRealms** [comment]
💡

I've just noticed, in my case at least, that if I turn off the preference named "Enable variable reference warnings" found in the Languages → GML Code section, that the IDE no longer stutters at all when doing code edits.  Even with Feather turned on.

I would suggest that the YoYo devs focus on that particular option as a potential lead to the poor IDE performance issues described in this bug report.

**Comment by Retrific** [comment]
I already had this setting disabled and still have the reported performance issues so I dont think this is on the right track. Good that it worked for you though

**Comment by ParodyKnaveBob** [comment]
Just more anecdotal evidence, sorry.

I've had [b]no[/b] discernable problems with stutter [b]until[/b] just today-or-so, coinciding around the time I implemented a constructor/new function pair for the first time. Obviously, could be unrelated, but worth a check I figure.

**Comment by craigtubb57** [comment]
MacBook Pro - 15-inch, 2018
OS: Ventura (but happened on versions prior)
CPU: 2.6 GHz 6-Core Intel Core i7
RAM: 16 GB 2400 MHz DDR4
GPU: Radeon Pro 560X 4 GB & Intel UHD Graphics 630 1536 MB

Still a major issue for me (v 2023.11.1.129). Tried:
- closing inspector
- closing all rooms
- turning off Feather
- upping Code Completion delay to 1000 (can it be turned off completely?)

None of this seems to work in the long term as after just 10-15 mins in a freshly restarted IDE the slowdown starts again.
I can see GameMaker itself sits at around 50% of a CPU on average, which is substantially higher than any other program I use.

Hopefully this helps the team.

**Comment by PrismaticRealms** [comment]
> I already had this setting disabled and still have the reported performance issues so I dont think this is on the right track. Good that it worked for you though

For clarity, I am still having performance issues, just no longer from making code edits.

**Comment by PrismaticRealms** [comment]
Some additional things I've noticed.  Sometimes when there are these little lags/stutters as described above, I see the following messages written to the macOS Console log:

default	15:09:43.478079-0400	spindump	GameMaker [12347]: spin: not sampling due to conditions 0x400000000
default	15:09:43.686880-0400	spindump	GameMaker [12347]: slow hid response: not sampling due to conditions 0x400208002
default	15:09:54.678580-0400	spindump	GameMaker [12347]: spin: not sampling due to conditions 0x400000000
default	15:09:55.427160-0400	spindump	GameMaker [12347]: slow hid response: not sampling due to conditions 0x400008002
default	15:10:03.427787-0400	GameMaker	Lost inheritances: {(
    <RBSInheritance| environment:(none) name:com.apple.launchservices.userfacing origID:210-153-112037 0>,
    <RBSInheritance| environment:(none) name:com.apple.frontboard.visibility origID:210-180-112039 0>,
    <RBSInheritance| environment:(none) name:com.apple.launchservices.userfacing origID:210-153-112037 0>
)}

And also, not necessarily occurring when there is a lag, but still concerning, is that sometimes I would see the following messages in the Console, as many as 70 of them in sequence:

default	15:11:38.995784-0400	GameMaker	MacOS error: -25294

Hope that helps.

**Comment by muddrox** [comment]
This is still a big problem for me and has been since December.  On top of the overall sluggishness, the debugger is the worst it has ever been and I never had any issues with it prior to December of last year.  Running my game in debug mode causes an assortment of stutters and other annoying instances of inexplicable slow down.

**Comment by Retrific** [comment]
Yea I hope someone is working on this ticket. This is super annoying and the biggest reason for why I rarely use Game Maker nowadays. It just takes the fun away. Game Maker was always the fast, lightweight engine that you can quickly start, working and iterate with. Having to restart the IDE every hour now and get annoyed by freezes and just general sluggishness and more and more slowdowns really kills this fun.

Please focus on this Yoyogames

@YYDan Any update on this?

**Comment by RefresherTowel** [comment]
> This is still a big problem for me and has been since December. On top of the overall sluggishness, the debugger is the worst it has ever been and I never had any issues with it before December of last year. Running my game in debug mode causes an assortment of stutters and other annoying instances of inexplicable slowdown. The debugger borders on being unusable at times. The stutters will occur in rooms where nothing is happening and they seem to occur in pattern-like intervals. It's obnoxious.

I'm not sure if we're thinking of the same thing, but I have had this wild amount of patterned stutter happen in my game while running in debug mode occasionally. It's more than enough to render the game actually unplayable (like, 1-2 second length pauses every 2-3 seconds). It seems to trigger if I alt-tab at the wrong time (basically, as the game window is being created at the end of the compile process, there's a very short window where if I alt-tab, I'll trigger this happening, though I can't get it to trigger on command, I've just noticed that whenever it accidentally gets triggered, it's when I'm alt-tabbing as the window is created).

Once it's triggered, it's there forever, your project is basically tainted. The "solution" I've been using is to export my project to a yyz, then import that yyz and make absolutely sure to save that new yyz imported project into a _different folder_ than your original project. If you import it and save it over the old project, this does not work (which makes me think there's something in the cache or settings or whatever causing the stutter that doesn't get overwritten in the import process, though clearing all the caches of the old project does not fix this). Once you've got your project imported to a new folder, the stutter should be gone. You can then go and (usually I save a backup before doing this), delete all the files from your old projects folder and copy + paste the new project files into that old project folder. It's a legit project destroying bug if you can't figure out the workaround, kinda nuts when it happens.

I've been meaning to get around to submitting this as a bug but have been swamped with work recently. I'll see if I can get to it over this weekend or something.

**Comment by muddrox** [comment]
> Yea I hope someone is working on this ticket (But since no one is assigned it doesnt look good for us). This is super annoying and the biggest reason for why I rarely use Game Maker nowadays. It just takes the fun away. Game Maker was always the fast, lightweight engine that you can quickly start, get something working and keep iterating with. Having to restart the IDE every hour now and get annoyed by freezes and just general sluggishness and more and more slowdowns really kills this fun.
> 
> Please focus on this, Yoyogames
> 
> @YYDan Any update on this?

I use Game Maker constantly but I have to agree here that this needs to be a priority.  It's genuinely painful to use Gamemaker right now and I am so confused as to why the dramatic shift in overall performance over the short span of a few updates.  I'm not a fast typer but game maker is struggling to keep up with what I type.  If yoyo games staff can't fix it right now then they need to tell us some workarounds or ideas to resolve it in the short term while we wait for an official fix.  Game Maker isn't in a good place right now and I think some of us would appreciate a bit more communication on the matter.

**Comment by Totobal5** [comment]
in the Beta IDE v2024.200.0.494 Beta Runtime v2024.200.0.511 the error still persist, 
~2 hours only using code editor.

**Comment by zreedy** [comment]
Hello everyone, as it relates to performances issues specifically in the Code Editor can you try these (very aggressive) options and report back if you see the microstutters decrease or disappear?
```
Enable variable reference warnings		OFF
Enable feather							OFF
Disable Right-To-Left Text Ordering		ON
Enable code completion					OFF
Undo stack							25
Syntax Checking delay (ms)				500
```

**Comment by Totobal5** [comment]
I set the options, do a reset and start working again (in the latest beta) and this is the ui.log

[ui.log](https://github.com/YoYoGames/GameMaker-Bugs/files/14315727/ui.log)

the stutter persist i do not feel that it less or more, it feels like before.

**Comment by johnson-jesse** [comment]
Please bump priority ASAP!

**Comment by konjerk** [comment]
This is driving me crazy at this point, completely demotivating. I tried moving Feather-related files out of the Game Maker app folder as some last resort, but nothing changed. Surprised the IDE even runs.

**Comment by nullbunny** [comment]
I wish a dev would update us on the status. It doesn't look like it is assigned to the upcoming milestone still which is very disappointing. I'm worried a fix is being worked on for 2022.2 at all since none of the betas so far acknowledged the issue. I really hope this doesn't mean we have to wait until the end of april or may for an attempted fix, because it's making my work days hell.

**Comment by craigtubb57** [comment]
I just wanted to mention some behaviours I do when using GameMaker which appear to contribute to the program slowing down. I've got no data on these, they are purely "feelings" I've accrued since about June 2023 when a recent update must have changed something.

Perhaps they're all misdirections, but maybe one of them will turn out to be an area of investigation yet to be explored. I'm sure the team is working on this difficult problem and they just need more data, but here's my observations such as they are:

- Saving manually very often (Ctrl / Cmd + S), pretty much after every change I make
  - I've been using computers since the late 90s so this is a habit I picked up from those uncertain times. I even know it saves when I run the game, but I do it anyway.
- Running the game very frequently, say once every minute or two
  - In my regular line of work I'm running unit tests this frequently, and the main app much less. But if I'm prototyping stuff in GameMaker I'm just making small tweaks and running the game. Very often. The CPU seems to creep up over time
- "Failed" runner termination
  - Not an error in game, or even an error in the runner, but very occasionally it seems like a process is left lingering from running the game. It's not displayed separately in any CPU monitoring tools, but feels like it's "bundled" with the main GameMaker IDE process.
  - This just seems to happen every now and then, but obviously more frequently the more I run the game (see above)
  - Maybe it is just some clean up that is taking longer than usual, but stacking due to frequency?
- Pressing "Stop"
  - Due to the point above, I find myself pressing the stop button after most runs as it feels like the game hasn't truly exited. My experience from last year was that running the game again without doing this multiplies the slowdown problem greatly, so it's become a new habit

My only other observation was that having Feather **ON** appears to improve performance. Again, no data on this, just a feeling. I've set the other settings as requested by zreedy above so will see how that goes.

Honestly though, I'm being pushed towards looking for an alternative editor. There are so many great ones that I use in my day job, and GameMaker feels quite out-of-date now. Is there any plan to incorporate an existing editor base e.g. Electron? Custom code editors seem to be dying in 2024, and for good reason.

Lastly a request: more communication please! Many people have mentioned this but just having a regular "we're still working on this, here's what we've eliminated, this is what we're trying" would go such a long way. It does feel like our concerns are being largely ignored right now, and that's pretty bad these days.
I mean, it's 2024, transparency is kind of standard now.

**Comment by konjerk** [comment]
I also save constantly, and run all the time. I've used tools in the past that have had awful issues so I have a habit. I love Game Maker for not having those problems, until now.

**Comment by CraterHater** [comment]
I am also experiencing this issue. Simply selecting something in the asset browser takes about half a second to update. This is incredibly frustrating. Anyone still any ideas?

**Comment by konjerk** [comment]
> Windows 10 Pro x64 Motherboard ROG STRIX Z390-F GAMING Intel(R) Core(TM) i9-9900K CPU @ 3.60GHz 3.60 GHz 32 GB RAM GeForce RTX 3090 Two monitors (3440x1440 and 1920x1080) Game Maker only used on the larger one, have had other monitors during the issue
> 
> I have always had Feather off, I disable whatever that right-to-left read thing is, I toy with time to check syntax or suggest code and it still happens in some similar way. I have many tabs open per Object, and pretty large window sizes too. I don't use Inspector really. I get stagger at editing anything in 2023.8 and 2023.11, and slowdown over time which has been an issue of different size since after 2022.4 I think.

So, I tried some more stuff. I installed the latest Game Maker version (2023.11 I guess) on a gaming laptop I had available. I then copied the Game Maker program folder from my workstation to it, as well as my project, and tried it all there. I still got lag that got worse and worse.

But then, I tried following the "completely uninstall Game Maker" guide on my main workstation, removing any temp files and registry data, then searching even more and finding some stuff. All deleted by me. I don't even save IDE Workspace settings that you can save out. I reinstall a supposedly fresh Game Maker on this PC, I do not reuse a single settings file whatsoever, and I do not load any saved workspace layouts. I do not change Feather settings, so this time it is on, since that's the default.

After about two days of not rebooting the IDE, and also working on my project (not constantly, I've not had time), I can't say I see too much difference in performance and lag. There's definitely a pause if I edit code then "drag the window around" to see if it staggers, but I think it's identical to when I launched this fresh Game Maker install. I can't say for sure it's fixed for me, but right now it is still running in a way I find fine. But again, I cannot say it won't start lagging, but it's certainly slower.

Recap:
- Complete unintall of Game Maker [using this](https://help.gamemaker.io/hc/en-us/articles/115002062812-How-to-perform-a-fresh-install-of-GameMaker)
- Find some remnant files still on my PC, could be old installs, but still delete them
- I do not make major Preference setting changes except for some font and code window sizes
- I do not put back in any IDE settings from previous install
- I do not try to import a workspace layout (I never did on my laptop where the lag happened though, so probably not relevant)
- Everything left as default as possible for now, except needed QoL for me
- Typing code, then moving a window within the syntax check delay, still staggers, but it doesn't seem to get much longer
- After two days of no reboot, and some active coding (a couple compiles too), it seems to not make any stuttering worse

Hopefully this could work for everyone, but this is my experience. Once again, maybe it will get laggy again later, and I'll get back on that, but it's way less frustrating right now. And Feather being on is honestly quite nice, haha. I hope it stays this way for me now. So, if there's an issue and this solves it, it's gotta be some legacy data in settings files, or a flag that doesn't quite recover right even if set back to off? Hard to tell.

**Comment by RhyminGarfunkle** [comment]
@konjerk Are you still seeing no stuttering? I've also had temporary success in the past with a clean reinstallation, but the lag came back within a few days.

**Comment by nullbunny** [comment]
@konjerk I tried that too a while ago. It got rid of the stutters and a different issue the team cannot reproduce where Undo and copy/pasting instances makes GameMaker go unstable. It works fine for a bit but after a week or so of working, the issues pop up again.

The project has gone through many GM versions and I suspect some old or leftover setting in there doesn't play nice with the new versions. :/

My next experiment is going to be to delete one file at a time from the AppData folders and take a not of when the issues disappear.

**Comment by konjerk** [comment]
> @konjerk Are you still seeing no stuttering? I've also had temporary success in the past with a clean reinstallation, but the lag came back within a few days.

I suppose I am feeling some chug sometimes now, in the end. The one benefit for me right now is that the Feather chugs don't eat my inputs very often when it lags, like it can with Feather off.

**Comment by RhyminGarfunkle** [comment]
How has everyone's experience with 2024.2 been so far? I've been using it for several days and haven't seen the lag at all. I spent a few hours working on a project, left GM open for over 24 hours, then came back to it, and after working for several more hours still no lag or stutters.

I don't want to say it's fixed, as the lag has gone away temporarily in the past, but that was always with fairly heroic measures such as a clean install. Has anyone else noticed an improvement just from installing the latest IDE version?

**Comment by konjerk** [comment]
> How has everyone's experience with 2024.2 been so far? I've been using it for several days and haven't seen the lag at all. I spent a few hours working on a project, left GM open for over 24 hours, then came back to it, and after working for several more hours still no lag or stutters.
> 
> I don't want to say it's fixed, as the lag has gone away temporarily in the past, but that was always with fairly heroic measures such as a clean install. Has anyone else noticed an improvement just from installing the latest IDE version?

I tried the beta and felt some lag, so I haven't installed this one for fear of compounding lag or something. Just waiting for this to be fixed at this point.

**Comment by Dyviik** [comment]
Still laggy for me.

**Comment by stuckie** [comment]
Reopening as it was erroneously closed due to automation.

Additional fixes merged in to do with opening/closing tool tips, and another one to do with the icon cache misbehaving.
We are still looking for additional performance issues, but as you can imagine, it's a big product and these are not in the most obvious places, nor do they necessarily match up to what you are seeing.

**Comment by Totobal5** [comment]
Still freeze on new Stable

**Comment by Grisgram** [comment]
From my observations of this phenonemon, I'd like to add, that it feels to me, as if it was attached to the debugger and runtime switches.

I can work without any issues for hours in GM, but after finishing the pure coding phase of a new thing, when I start testing it, GM runs into these freezes very quickly.
Switching between windows and html and sometimes starting with F5 and sometimes with F6, for the debugger.
After... like 3 or 4 sessions with a debugger, then switch to html, switch back to win, this time without debug, then again win with debug and so on... quite "normal" debug/bugfixing/development behavior.

Maybe there are leftovers, when you "stop" the game using the different ways avaiable:
- the stop button in the main toolbar
- the X button in the debugger tab
- clean program exit in the game due to some "quit" button in the game

if you combine those wildly, stuttering quickly starts rising. maybe, the debugger X button has leftovers compared to the stop button in the main toolbar? or the debugger has a left over when the game exits normally?

and it is also definitely tied to the number of Ctrl-Z / Ctrl-Y you are using. Undoing larger series of edits still lets the memory consumption of GM rise.

**Comment by bsabiston** [comment]
Is anybody working on this?  IDE becomes unusable after an hour.  It's been like this for a year or more, as long as I can remember

**Comment by stuckie** [comment]
Yes, we are still working on this.
There have been many other performance fixes, even just recently:
https://github.com/YoYoGames/GameMaker-Bugs/issues/4887
https://github.com/YoYoGames/GameMaker-Bugs/issues/2503

The main issue is that for each one we have to be able to:
- Get a consistent reproduction case
- Figure out what is causing the problem
- Fix it
- Then test it against the reproduction case

In many cases, the problem is getting that consistent reproduction case as there are many systems within GameMaker that could be interacting, where some can be turned off, or tuned to different preferences.

As a case in point, I have just come back from some annual leave and I had left GameMaker running, connected to a profiler, with a large project open with many windows... upon my return, it was using less than 1Gb of memory and was running fine.. so the combinations of preference settings, project make-up, and general actions makes this tricky to pinpoint, and we do feel as frustrated as you do with playing whack-a-mole with several disparate issues, and not one real cause at fault.

**Comment by nullbunny** [comment]
@stuckie Thanks so much for the update! If it helps at all, the project I submitted for #4786 has these issues, too. Hope it helps recreating them.

You might have to edit around a little bit, save a bunch of times, and copy/paste undo instances in rooms a bunch.

**Comment by stravee** [comment]
Someone who can reproduce the issue regularly should offer the Gamemaker team access to their PC via teamviewer or similar, so they could get a good look at the issue while it is happening.

**Comment by skyelynwaddell** [comment]
sames

**Comment by kt-altyn** [comment]
i am party of this gang by today. i have more than tens of thousand line of codes, 16 code windows open. started to lag so bad. like every 2-3 second, ide freezes and i can't work on my project this way. would closing opened windows or splitting one function into multiple ones solve the problem?

**Comment by BPOutlaws** [comment]
**Found a fix for anyone having this problem to try:** I just found out that if I run GameMaker, then go to my Task Manager, then the Details tab, then right-click on GameMaker.exe, choose "Set Affinity" and de-select almost all my 8 cores so only 1 or 2 of them are selected, it no longer does this lagging GPU-maxing melting my computer overheating thing. Only downside is you have to do this every time you start the GameMaker application up because it'll default to using all your cores each time.

Hope this bit of info helps YoYo narrow down what could be causing this.

**Comment by BPOutlaws** [comment]
More details:

Every 15 min or so the IDE just suddenly starts lagging and maxing out my GPU (specifically the "3D" metric). I've got a Microsoft Surface Pro 8 with 16GB of GPU memory available (Intel Iris Xe Graphics) and I don't have Feather or any other frills on and I close/never use the Inspector, and rarely use the room/image editors. I mainly just have a single code editor window open with <6 tabs open. My project is new so it's very small (1 sprite, 1 room, 2 objects and a handful of scripts) and I frequently save and test (no debugger). I've been using the exact same device for a couple years now and I know that a year ago it didn't do this. Clearing out the cache's etc doesn't help, it just inevitably ends up randomly deciding it needs 100% of my GPU for a few min while just displaying the same code window I've been typing in for the previous 5 min.

Once it starts lagging, if I keep my Task Manager performance tab open to show the GPU usage, and if I minimize the GM IDE the GPU usage (again, specifically the "3D" metric) immediately drops down to 20%, but then if I maximize the IDE it skyrockets to 100% GPU usage and lags.

It eventually settles back down after like 5 min or so, or I can restart GM but inevitably it ends up doing it again.

> with a large project open with many windows... upon my return, it was using less than 1Gb of memory and was running fine..

In my case it's not that it lags infinitely, it just goes through these random (ie - while I'm just in the middle of typing code) massive long periods of maxed out 100% GPU use that settle down after like 2-5 min but my entire computer is basically unusable until it's done. If I minimize the IDE window the GPU use immediately plummets back down to a sane level. It doesn't seem to be triggered by a specific action (ie - I click a button or do an Undo or something that immediately makes it start lagging), it seems to be something GM is doing or collecting in the background until it suddenly starts to chug.

Wish I could just use GMEdit but it looks like the latest GMS2 builds broke GMEdit so that's out :/

**Comment by ThylacineStudios** [comment]
I just realized that when the lag occurs for me, I can pan over to a different code editor of a different object and it goes away completely. Then, if I pan back to the original one, it starts lagging again. Restarting the IDE fixes everything until one starts randomly lagging again.

**Comment by NanoSharkGames** [comment]
According to Activity Monitor, after a couple hours of using the Mac IDE in 1 session, a lot of ports are being used by the IDE and could be a result of a thread leak (according to @stuckie).

Hopefully this helps to pinpoint the cause(s) of the issue.

**Comment by skyelynwaddell** [comment]
you need to switch to the windowed code editor and then make sure you dont constantly have 1000 scripts open at a given time, and keep your workplace clean and close objects/tabs when done, ever since doing this no gm lag.

**Comment by skyelynwaddell** [comment]
> you need to switch to the windowed code editor and then make sure you dont constantly have 1000 scripts open at a given time, and keep your workplace clean and close objects/tabs when done, ever since doing this no gm lag.

preferences > open script, object events, event scripts in same window check marked

**Comment by YYDan** [comment]
Adding the ui.log from https://github.com/YoYoGames/GameMaker-Bugs/issues/4387

https://api.gamemaker.io/api/github/downloads/fe8c6c41-20bb-40bf-825c-1d55345deb82

**Comment by Totobal5** [comment]
In IDE v2024.4.0.137  Runtime v2024.4.0.168 the code editor lag is worse that the previous montly version.

**Comment by NanoSharkGames** [comment]
Same as above, even when using editing code in fullscreen. Though, I use the object code editor as is (not fullscreen).

**Comment by Grisgram** [comment]
I also think, the keyboard input lags are back -- my keyboard macros sometimes destroy the code when gms does not executes the keystrokes in time...

**Comment by ThylacineStudios** [comment]
Same as above, not only is the lag the same as before, but sometimes my keyboard input is ignored now.

**Comment by TheoryGeorgiou** [comment]
> In IDE v2024.4.0.137 Runtime v2024.4.0.168 the code editor lag is worse that the previous monthly version.

Yeah I haven't updated waiting to see if this issue is resolved or made worse. looks like I have my answer. 
This is literally the only issue or feature I care about for game maker. This might be the last game I use game maker for if this isn't resolved by the time i'm done with my current project. and I've been using game maker for 20 years. 

I hate this bug and am really frustrated at the lack of urgency by game maker staff. Can we fork game maker and I will PAY SOMEONE to fix this issue. not joking.

**Comment by Totobal5** [comment]
> I also think, the keyboard input lags are back -- my keyboard macros sometimes destroy the code when gms does not execute the keystrokes in time... especially newline and tab-characters

Same here, likely the changes on keyboard inputs come with a bug.

My UI.log from a session only code editor.

[ui.log](https://github.com/YoYoGames/GameMaker-Bugs/files/15211638/ui.log)

**Comment by TheoryGeorgiou** [comment]
@YYDan Can we please take this issue seriously, it's been almost 6 months. this is y our most commented on issue out of 3,000+. 
"moved this to Backlog in [Team Workload]" doesn't look great.

**Comment by RhyminGarfunkle** [comment]
Something weird I've noticed about the lag: I saw huge improvements with 2024.2, but only for _new projects_.

It doesn't seem to have anything to do with asset count, I've imported collaborator's YYZ projects with large numbers of assets into a new project and worked on it for days without seeing any lag. However, once I open up a project that was created in a version pre-2024.2, specifically a .yyp file in a local project directory, I can only work for 30-60 minutes before the lag shows up.

I'm trying an experiment now, exporting a project that I know causes lag to YYZ then re-importing it to a new project, to see if that does anything.

**Comment by stuckie** [comment]
We are still looking into this.. and we do fix performance issues as we find them, particularly as this overall "performance issues with the IDE" ticket has taken us all over the place.
As I've stated in one of the other performance issue tickets, this has not been an easy one to track down as it seems to be dependent on a whole range of things; and even the comments in here are for different areas of the IDE, and don't agree on one specific thing.

For some transparency, we have only recently confirmed there is a slow mach port leak on macOS.. we don't appear to be leaking within the .NET area, but we are leaking something in either our interop code, or with a native library, and are now tracking that down. We have not been able to reproduce this until recently.

The keyboard issues are also a separate thing, but seem to have been made worse recently, likely just due to combinational explosion from other work happening at the same time.

These will be fixed. And, in some way of a silver lining, as they are getting more visible, they should be easier to track down. Especially as we have now, finally, been able to reproduce, and confirm, that *something* is leaking...

I can only apologise for the time it's taken to resolve these. We absolutely feel your frustrations, as they've been worse for us in that we've not even been able to confirm what you have been seeing for the most part! We can't fix something we can't see. Hopefully, this leak we are seeing now is the final culprit, and we'll have better news soon.

**Comment by ThylacineStudios** [comment]
Thank you for the update! Looking forward to seeing this fixed once and for all.

**Comment by TheoryGeorgiou** [comment]
> We are still looking into this.. and we do fix performance issues as we find them, particularly as this overall "performance issues with the IDE" ticket has taken us all over the place. As I've stated in one of the other performance issue tickets, this has not been an easy one to track down as it seems to be dependent on a whole range of things; and even the comments in here are for different areas of the IDE, and don't agree on one specific thing.
> 
> For some transparency, we have only recently confirmed there is a slow mach port leak on macOS.. we don't appear to be leaking within the .NET area, but we are leaking something in either our interop code, or with a native library, and are now tracking that down. We have not been able to reproduce this until recently.
> 
> The keyboard issues are also a separate thing, but seem to have been made worse recently, likely just due to combinational explosion from other work happening at the same time.
> 
> These will be fixed. And, in some way of a silver lining, as they are getting more visible, they should be easier to track down. Especially as we have now, finally, been able to reproduce, and confirm, that _something_ is leaking...
> 
> I can only apologise for the time it's taken to resolve these. We absolutely feel your frustrations, as they've been worse for us in that we've not even been able to confirm what you have been seeing for the most part! We can't fix something we can't see. Hopefully, this leak we are seeing now is the final culprit, and we'll have better news soon.

Thanks for the update, sounds like a lot more than just this is wrong with the IDE tons to get sidetracked with. 
I have been having the stuttering, large projects from using the IDE after a few hours on window 10 for a long time now not macOS, so that might be a different issue.

I appreciate the update but it does not sound very hopeful for Game maker's future. Especially hearing how it's worse in the beta. Been using GM for 20+ years and this is the first issue that has me looking towards using another engine. It breaks my heart.

**Comment by konjerk** [comment]
> I appreciate the update but it does not sound very hopeful for Game maker's future. Especially hearing how it's worse in the beta. Been using GM for 20+ years and this is the first issue that has me looking towards using another engine(there has been other issues, but those were fixed in a reasonable time). It breaks my heart.

Lamenting and goading like that won't speed it up somehow, I feel they made it clear what the struggle has been. All that kind of language you used can achieve (if anything) is upset people.

Glad the GM team have found some reproducable things! Good luck. My issues have gotten somewhat better with each update, but hopefully it can get to zero soon!

**Comment by rwkay** [comment]
> > Glad the GM team have found some reproducable things! Good luck. My issues have gotten somewhat better with each update, but hopefully it can get to zero soon!
> 
> What?! My IDE performance since the latest patch is the worst it has ever been. I start Game Maker and immediately it feels laggy, stuttery and barely usable. Immediatly! Even with a blank worspace. Shortcuts are not recognized. Stutters every few seconds even when I don't even do anything. Just a huge pain to use. It's been so long since we reported the first performance issues and there were many updates since then but in my experience it only kept getting worse.
> 
> I would love to be as hopeful as you are but I am at a point where I seriously consider leaving GM for good. Have I not spend years and years building up libraries I would've moved on already. Solely because of the performance issues that Yoyogames cant seem to fix. Even if they fix one issue the next new feature will introduce two new ones. I don't see this ever getting fixed unless they change their plans big time. We need a miracle.

What platform are you running on... 

see my comment on https://github.com/YoYoGames/GameMaker-Bugs/issues/5796#issuecomment-2100185077 we need a lot more info... if we are going to see these issues finally resolved.

This is not something we are seeing on the equipment here but show us what you are seeing, we have improved performance in many places throughout the IDE but obviously not the right things for your situation... show us what is happening.

Russell

**Comment by Retrific** [comment]
I have sent you my info multiple times already. You probably know more about my machine than I do myself.
[DxDiag.txt](https://github.com/YoYoGames/GameMaker-Bugs/files/15247424/DxDiag.txt)

**Comment by Retrific** [comment]
The project does not seem to matter. I also feel the lag in the template projects. But it's difficult to capture, it's more something that you need to feel. It's a lot of micro stuttering and your cursor getting stuck for half a second or something and most recently it's the shortcuts not being recognized which makes this feeling a lot worse. I can try to capture the lags on a video today but it will probably not be very visible when you don't feel it yourself and might not even be recordable at all when it also affects the capturing software. We'll see.

It's not that GM is literally unusable with these lags, it's just not fun to use because it feels so sluggish and unresponsive and you constantly feel like the IDE is going to crash soon or your machine is going to catch on fire. But then when you switch to any other software it all works perfectly fine. That's how you know it's GM that is struggling, not your PC. And when you switch back to GM you are immediatly greeted with the lags again and it kills your motivation to keep working. Especially since it gets worse the longer you have it open.

This is also my ticket: https://github.com/YoYoGames/GameMaker-Bugs/issues/5752
And here is my ui.log from just now
[ui.log](https://github.com/YoYoGames/GameMaker-Bugs/files/15247535/ui.log)

**Comment by Retrific** [comment]
Btw I am using a 144Hz Monitor and have also set my workspace framerate to 144. So this might explain why I am more sensitive to lag spikes or low framerates.

Maybe there is some code in GM that targets 60fps for some synchronous background task???

Doesnt explain the lag spikes but still something worth investigating.
I really want the IDE to run at my native refresh rate or it will always feel sluggish.

**Comment by TheoryGeorgiou** [comment]
> ^^^ Defeatist attitudes helps nobody and speeds up nothing.
> 
> Glad the GM team have found some reproducable things! Good luck. My issues have gotten somewhat better with each update, but hopefully it can get to zero soon!

I've been optimistic for 6 months dealing with this time consuming bug with no progress from GM team. Any senior dev can see the writing on the wall, they are pumping out big changes/features without keeping up with issues. That is not maintainable.

**Comment by Retrific** [comment]
The mouse stutter issues are new actually, I have not experienced them this badly in older versions of GM. It started with the latest patch that I have them pretty much constantly even after just restarting the IDE. So it's not the same issue as the workspace stutter that I described when I opened this thread.

The difference is that now my mouse cursor gets stuck when before it was the view of the workspace that got stuck, and only after typing or editing an asset. Now the mouse getting stuck happens every time I move the workspace, doesn't matter if I edited any asset.

**Comment by Retrific** [comment]
Here is a video of this new mouse stutter that I recorded with my phone. Imo you can see it quite clearly. Whenever I let go of the MMB the cursor stops moving smoothly for a moment and instead "stutters" before resuming to being smooth.

https://github.com/YoYoGames/GameMaker-Bugs/assets/8227911/b1dd3b08-9954-42eb-8134-c6051c834a2b

**Comment by Retrific** [comment]
Actually found something about the mouse stutter. I fear this thread will go off-topic though since the performance issues described earlier are probably not related to the mouse stutter, so I will continue posting in the closed issue thread and suggest to re-open it.

**Comment by BPOutlaws** [comment]
> Actually found something about the mouse stutter which I posted in the closed thread of #5824

From that thread rwkay wrote:
> "They are all related, we have found that the IDE was using a default setting which meant that it was waiting for vsync rather than displaying immediately (which is the correct setting) this has been fixed on the internal build and will be part of 2024.6 Betas - this is all part of https://github.com/YoYoGames/GameMaker-Bugs/issues/2528 and should address the issue that you have found - we will not be reopening this bug"

Does that mean this bug is finally going to be fixed??? GMS is literally unusable for me at this point, I took some time off and worked on other projects and coming back is just a nightmare of frustration. I can't get any actual coding done because I'm constantly fighting the IDE and long periods of 100% GPU usage.

**Comment by Retrific** [comment]
If you happen to use an Nvidia GPU and/or a GSync Monitor you should try my workaround that I described in issue #5824 

I have this workaround applied for two weeks now and since then have not experienced any lag in the latest version of Game Maker 2024.4.1. No mouse stutter but also no workspace movement lag after writing code.

> My new workaround: Disabling G-Sync for GameMaker using the Nvidia control panel. However, doing this caused screen flicker issues (where the screen turns black randomly for a brief moment while GameMaker is running). But I was able to fix that too by changing back the refresh rate and vsync setting in the control panel (which auto-changed when disabling G-Sync) - This caused the G-Sync setting to also turn back on, but apparently, this is just a visual glitch in the control panel because the mouse stutter was still fixed, so I assume G-Sync is still disabled.

**Comment by shamusl** [comment]
The GM IDE performs very poorly performance-wise on a very high end macOS system while the system is not overloaded. It’s the only application on my system that slows down the longer it runs, and this is something that is getting worse as the months go on and features are being added. It seems every release is slightly worse, or, at least, just as bad. 

There can be 500ms times between hitting a key on the keyboard and text entry. This happens in large projects with lots of code and I do not have Feather enabled either.

The only fix is to close the IDE and re-open it, but with 70K+ lines of code, I need to do this maybe every 2-3 hours of running the IDE. It is clearly worse as the project complexity increases.

**Comment by stravee** [comment]
There has been talk that these freezes and lags are due to the old code editor and switching to the new one (which is at this time not yet production ready) would eliminate them entirely.

Has any of you tried the new code editor in the beta releases?

**Comment by shamusl** [comment]
> There has been talk that these freezes and lags are due to the old code editor and switching to the new one (which is at this time not yet production ready) would eliminate them entirely.
> 
> Has any of you tried the new code editor in the beta releases?

The latest beta and new code editor does not fix this issue. Both the latest stable and the Beta are both leaking Mach Ports and leaving threads open endlessly on macOS at the same rate. This is causing thrashing with the kernel. After a while, it starts to impact overall system stability. I’m filing a feedback with Apple, hopefully they read it and add new protections to macOS.

If anyone at YoYo reads this *please* contact Apple and Microsoft developer support! They want to see issues like this! You are an important developer to Apple they will be happy to meet with you at WWDC or elsewhere. Sending a team member in person or one-on-one via video and laying out this issue might be the only way this is ever found.

If something is wrong with Mach or .Net both of those companies want to know that!

**Comment by stuckie** [comment]
2024.04.1 definitely had a fix for macOS that should have cleaned up those mach ports - there was an issue where we hadn't cleaned up some interop objects when checking for a battery, which was further aggravated when the application was minimised.

We've also just found another hiccup where the source control implementation was trying to do a status check on every project save - that should be in the latest 2024.600 beta.

We are still hunting these down, but just these two alone aren't in obvious places, so it's slow going.
I've also managed to get an Ubuntu setup where the IDE is really struggling for some reason, so I'm hoping to be able to profile that soon.

**Comment by shamusl** [comment]
> 2024.04.1 definitely had a fix for macOS that should have cleaned up those mach ports - there was an issue where we hadn't cleaned up some interop objects when checking for a battery, which was further aggravated when the application was minimised.
> 
> We've also just found another hiccup where the source control implementation was trying to do a status check on every project save - that should be in the latest 2024.600 beta.
> 
> We are still hunting these down, but just these two alone aren't in obvious places, so it's slow going. I've also managed to get an Ubuntu setup where the IDE is really struggling for some reason, so I'm hoping to be able to profile that soon.

This is good news. Thank you! I’ve been seeing peoples reports of this bug being filed as Not Doing or Done over and over again and that was not instilling confidence.

That said, Mach ports are still leaking along with memory consumption climbing. I don’t use feather or the built-in source control. The number of ports open at any time is often over 20,000 which is higher than any application available for the OS. Profiling the IDE using the macOS built-in tools also reveals threads that are opened, given work, and then never closed regardless of closing editors, documents, workspaces etc.

Memory pressure on the system is low, and I am not swapping, so why would the (very serious) memory leak be causing performance issues, if it hasn’t exhausted resources yet? 

The CPU and GPU are almost entirely idle as well, with the IDE only taking up 60%-ish, leaving the other 940% of the system free.

**Comment by stuckie** [comment]
Just a further update to this to keep people in the loop, I now have a macOS setup that exhibits very poor performance with the IDE.. so we're finally starting to close in on this.

**Comment by NanoSharkGames** [comment]
> Just a further update to this to keep people in the loop, I now have a macOS setup that exhibits very poor performance with the IDE.. so we're finally starting to close in on this.

That's great to hear! Would optimizations be included in a potential fix? GM uses substantially more power on my laptop than 2024.2. Maybe it's also because of the ProMotion refresh rate adjustment bug.

**Comment by shamusl** [comment]
> > Just a further update to this to keep people in the loop, I now have a macOS setup that exhibits very poor performance with the IDE.. so we're finally starting to close in on this.
> 
> That's great to hear! Would optimizations be included in a potential fix? GM uses substantially more power on my laptop than 2024.2. Maybe it's also because of the ProMotion refresh rate adjustment bug.

I agree with you so much! I really hope they’re closing in on this bug like they said.

That said, to add additional feedback to your comment I can confirm that this problem happens whether I am using my Studio Display (5K@60hz) or on the built-in ProMotion display (120hz, native 16 inch M1 Max MacBook Pro display) so I don’t think that the problem is ProMotion in particular.

I’d also really like to back up the statement about high energy usage. GM IDE consumes 1000+ in the Energy Monitor panel of the Activity Monitor. This causes the battery life to drop about 0.5-1% charge per minute when the laptop normally lasts 12 hours with relatively heavy usage (Xcode, iOS simulators, hundreds of tabs). 

GM IDE uses the most energy of any application I’ve ever seen. SoC wattage jumps from 8 watts all the way up to 30 watts, even when the IDE is not visible/hidden. Even playing Resident Evil Village consumes less power than the GM IDE. 

This problem has gotten worse as time has went on and the latest release addressing some of the leaks does not address this at all. I would hope that addressing the performance issues will also address the high energy utilization as well.

**Comment by AvioxArcade** [comment]
Hoping to jump into this conversation and provide some value. 
I've been dealing with stutters for a while now, and I'm trying to discern when and where stutters are actually happening. 

When analyzing my cpu usage with the Resource Monitor, I'll see a spike in one CPU when the stutter happens, while overall (and the other CPUs) all seem fine:

![image](https://github.com/YoYoGames/GameMaker-Bugs/assets/61194078/50998395-0825-4a1d-99ae-88252c86e238)

I typically have 3 or 4 tabs open. one is a workspace, and others are code windows with several different files open in each.
Turning feather off seems to lessen the impact, but doesn't remove it.

**SYSTEM SPECS:**
**OS:** Windows 11 Home 64-bit
**Processor:** AMD Ryzen 7 5700G (16 CPUs, ~3.8GHz)
**Memory:** 32,768 MB RAM
**Graphics:** NVIDIA GeForce RTX 3060 Ti (8024 MB VRAM)
**Primary Display:** 5120x1440, 165Hz

I see most stutters when coding (maybe because it's what I do most). They usually last about 1 second. 

I also see a lot of slowdown when placing or removing objects in the room editor, often when the top right status bar says "analyzing project" or similar

The most reliable slowdown has been from using the "close all but this" (or similar) option in a workspace when I have a lot of assets open. Sometimes the "close all" task will freeze the IDE for several seconds. I do have IDE animations turned on, not sure if turning them off would help (but usually the freeze happens at the end of the animation, where the window is a fairly flat box)

**Comment by AvioxArcade** [comment]
On second look, it looks like my SSDs might be involved as well.
When clicking into gamemaker after clicking away, the IDE hung for about 1.5 seconds, which showed up on the resource monitor as the spike on the right side of the graph:
![image](https://github.com/YoYoGames/GameMaker-Bugs/assets/61194078/a455f297-9a1f-4e6d-abb6-b4d3335c3f70)

(everything on the left side was because something was downloading. I paused the download around the halfway point)

**Comment by jonathan-markland** [comment]
After some consultation with Zach Reedy I am going to tentatively say we have a fix for this. @YYDan @stuckie

**Comment by stuckie** [comment]
For those waiting patiently, the above potential fix for this has gone out in 2024.800.0.597, which has just been released.

**Comment by jonathan-markland** [comment]
I believe this is fixed now.

**Comment by nullbunny** [comment]
I tested the beta for a couple of days now and so far the issue hasn't popped back up. Thank you for all your hard work, I am so happy!! Working in GameMaker is a joy again. <3

**Comment by stuckie** [comment]
Excellent!

Thanks for letting us know... fingers crossed that's the end of it!

**Comment by craigtubb57** [comment]
I'm sorry to say the latest beta release appears to make no difference for me.
Strangely, it initially appeared to just make the IDE much slower in general. After a bunch of restarts because the program became unusable, it's right back to the problem I had before.

I'm also not convinced it's a problem you can see in the Activity Monitor/Task Manager, and "feels" more like the program is collecting a list of files/game objects, and analysing them every time you make a code change. The list seems to continually grow, perhaps containing unnecessary duplicates and analysing the same file/object multiple times?

Nothing scientific there, just a "gut feel".

I say this because the `Analysing Project` loading bar appears in the top-right very often and after some very simple actions on my part e.g. Saving the project, closing a window
It appears to show for longer and longer periods of time, and it seems the more often it appears the slower the program gets. It quite quickly (15 mins sometimes) gets to the point where it cannot cope with me clicking on a position in a code editor (it takes a while to think and nothing happens).

In terms of my usage of the program, I notice the degradation happens quicker, the more frequently I do the following actions:
- save
- run a game
- reload the project

Hopefully that's all useful information.

---

IDE Version: `Beta 2024.800.0.609`
Current Runtime: `Beta 2024.800.0.633`

`MacBook Pro - 15-inch, 2018`
OS: `Ventura 13.6.9` (but happened on versions prior)
CPU: `2.6 GHz 6-Core Intel Core i7-8850H`
RAM: `16 GB 2400 MHz DDR4`
GPU: `Radeon Pro 560X 4 GB & Intel UHD Graphics 630 1536 MB`

---

I admittedly haven't tried toggling various settings e.g. Feather like I have with previous versions, but should I really have to? This problem has been labelled "fixed" in the current beta so I take that to mean it will work out-of-the-box.

I'm sure this is a difficult one to pin down and I'm very appreciative of all the time and effort you folks are putting in to get to the bottom of it.
Unfortunately it currently makes GameMaker unusable :(

**Comment by nullbunny** [comment]
Oh man.. If the bug is fixed for me and others, hopefully it means your project is just another edge case that's missing in the fix. Have you tried to completely reset GameMaker with the procedure that @konjerk suggested earlier on in the thread?

**Comment by stuckie** [comment]
Yes, this one has been full of edge cases... @craigtubb57 could you submit a new ticket and include your project so we can have a look?
All files uploaded via the IDE's bug reporter remain private and only accessible to GameMaker staff, as we'd likely need the full project to drill down on this.

Thanks!

**Comment by YYDan** [comment]
We have not seen any issues and the overwhelming community feedback is that this has been resolved for several weeks (bar the one Mac IDE report above, for which a dedicated new issue has been requested). so cautiously closing this one now.
